### PR TITLE
Adding Qwen3-Next SFT recipe

### PR DESCRIPTION
**Pull request by adityavavreNVDA**
# What does this PR do ?

This PR adds a Full-SFT Recipe for Qwen3-Next and an example finetuning script.

Following are the SFT training curves on the Squad data:
- Validation Loss is at 0.18 at iter 90 which meets the requirements.
- Resume at iter 100 matches the first run exactly.
- Model at iter 100 generates coherent outputs: 
======== GENERATED TEXT OUTPUT ========
Prompt: Hello, how are you?
Generated: Hello, how are you? I'm a little confused about the difference between "I have been to" and "I have gone
=======================================

W B Chart 11_6_2025, 1_36_00 PM
W B Chart 11_6_2025, 1_35_46 PM
W B Chart 11_6_2025, 1_32_06 PM
W B Chart 11_6_2025, 1_36_30 PM

### CODE REVIEW

**Comment by adityavavreNVDA** [comment]
/ok to test [e2ad8e6](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/e2ad8e6d9df7aaad5d78bd5c98be201be341511d)

**Comment by adityavavreNVDA** [comment]
/ok to test [0b0322d](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/0b0322d8fe1e7545c7e7b1d1e854a7f7ce360edd)

**Comment by adityavavreNVDA** [comment]
/ok to test [dbe0f0f](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/dbe0f0fdb410f61526eda7a8dc45a4458383c932)

**Comment by adityavavreNVDA** [comment]
/ok to test [9a09229](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/9a0922957a64e494e58e62ad7c04a8b05951358a)

**Review by yaoyu-33 — APPROVED** [review]

**Comment by adityavavreNVDA** [comment]
/ok to test [d51ac43](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/d51ac4302785da52592d2189fb24bb3accf25d7e)

**Comment by adityavavreNVDA** [comment]
/ok to test [ea0230f](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1215/commits/ea0230fac0254d91b4fae32282d28b0997f51e48)
